contrib/aero-snap/main.cpp

```
@@ -0,0 +1,560 @@
+            current.x - extents.topLeft.x,
+            current.y - extents.topLeft.y,
+            current.w + extents.topLeft.x + extents.bottomRight.x,
+            current.h + extents.topLeft.y + extents.bottomRight.y,
+        };
+        const double   anchorX = std::clamp((cursor.x - currentOuter.x) / std::max(1.0, currentOuter.w), 0.08, 0.92);
+        const double   anchorY = std::clamp(cursor.y - currentOuter.y, 0.0, std::max(0.0, currentOuter.h * 0.25));
+        const Vector2D restoredOuterSize{
+            restoreBox.w + extents.topLeft.x + extents.bottomRight.x,
+            restoreBox.h + extents.topLeft.y + extents.bottomRight.y,
+        };
+        CBox anchoredRestore{
```

Clang (14.0.6)

[warning] variable 'anchoredRestore' is not initialized

(cppcoreguidelines-init-variables)